sale_order_revision/view/sale_order.xml

```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+    <data>
+        <record id="sale_order_form" model="ir.ui.view">
+            <field name="name">sale.order.form</field>
+            <field name="model">sale.order</field>
+            <field name="inherit_id" ref="sale.view_order_form"></field>
+            <field name="arch" type="xml">
+                <notebook position="inside">
+                    <page string="Revisions">
+                        <field name="old_revision_ids" >
+                            <tree>
+                                <field name='name'/>
+                                <field name='create_date' string="Superseeded on"/>
+                                <field name='create_uid' string="Superseeded by"/>
+                                <field name='state' invisible='1'/>
+                            </tree>
+                        </field>
+                        <group>
+                            <field name="current_revision_id"/>
```

Comment:
there is always the possibility of using advanced search. 

I updated the view to only display the field if the SO is not active (and hence not the current revision).